students/lakatorz_izaak/lesson_04_unit_testing/practice_projects_two.py

```diff
@@ -0,0 +1,45 @@
+# Write test module with 3 unit tests for distance() from The length of the
+# segment exercise from previous exercise Snakify / Lesson 8 / Problems:
+
+import length_of_segment
+import pytest
+
+
+def test_if_raises_type_error_for_none():
+    with pytest.raises(TypeError):
+        length_of_segment.distance(None)
+
+
+def test_if_raises_value_error_for_string():
+    with pytest.raises(ValueError):
+        length_of_segment.distance(1, 0, 1, 'aoeu')
+
+
+def test_corner_cases():
+    # Zero length
+    assert length_of_segment.distance(3, 5, 3,
+                                      5) == 0, 'Non-zero value, should be ' \
+                                               'zero. '
```

Comment:
Zerknij sobie na zalecenia PEP8 odnośnie wcięć (konkretnie `# Hanging indents should add a level.`):
https://www.python.org/dev/peps/pep-0008/#indentation
Łamanie stringów jest bardzo mało przyjazne dla oka, dlatego zalecam zmianę.

Podejście 1:
```
assert length_of_segment.distance(
    3, 5, 3, 5) == 0, 'Non-zero value, should be zero.'
```
Podejście 2:
```
distance = length_of_segment.distance(3, 5, 3, 5)
assert distance == 0, 'Non-zero value, should be zero.'
```

Zalecam poprawkę, najlepiej na opcję 2, gdzie w ogóle nie ma łamania linii czym znacząco zyskujemy na czytelności. Przejrzyj również pozostałe przypadki pod tym kątem.